### PR TITLE
Enable sdcmonitor

### PR DESCRIPTION
**Pull request by kumarkgosa**
# Description
Adding support to sdc-monitor for powerflex driver through operator.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/477|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# Screenshots of sdc-monitor container: 
The driver is working fine without any restarts with sdc-monitor enabled :

Capture

The container sdc-monitor is enabled when described worker node.

sdc-monitor

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by bharathsreekanth** [comment]
Update with GitHub issue #.This is part of CSM operator feature, so you can include that.

**Comment by bharathsreekanth** [comment]
Also, can you add some way for others to know that the SDC monitor side-car was deployed in the PR?

**Comment by JacobGros** [comment]
can we add unit tests for the new code added?

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]
We should try to get coverage back up to ~95%, which is where it was before this PR. I can work with you on this, hmu

File: pkg/drivers/powerflex.go [hunk 1]
```diff
@@ -185,3 +203,31 @@ var (
 func IsIpv4Regex(ipAddress string) bool {
 	return ipRegex.MatchString(ipAddress)
 }
+
+// Getmdmipforsdc - fetching mdmip from sdc initcontainer and passing it to sdc-monitor
+func Getmdmipforsdc(sidenv []corev1.EnvVar, mdmFin string) (corev1.EnvVar, error) {
```
[the remark this answers is not in the document]
Comment (kumarkgosa) [reply pkg/drivers/powerflex.go]: Changed it to GetMDMIpForSdc jooseppi.. Thank you

**Comment by JacobGros** [comment]
Are we able to disable the sdc-monitor car?
I see in samples, 
https://github.com/dell/csm-operator/blob/8fe5a4c8bd2933bcf3d44ee0b133f01b74b595e4/samples/storage_csm_powerflex_v240.yaml#L41 

but in screenshot its enabled

**Comment by JacobGros** [comment]
2.3 and 2.4 should both support enable/disable sdc-monitor
can we add the enable/disable flag to this section: https://github.com/dell/csm-operator/blob/8bb767883348cbfe4de48c79abcab74dc74dc069/samples/storage_csm_powerflex_v230.yaml#L38 as well?

**Comment by kumarkgosa** [comment]
> Are we able to disable the sdc-monitor car? I see in samples,
> 
> https://github.com/dell/csm-operator/blob/8fe5a4c8bd2933bcf3d44ee0b133f01b74b595e4/samples/storage_csm_powerflex_v240.yaml#L41
> 
> but in screenshot its enabled

Yes, we can disable/enable it manually. By default it will be disabled.

**Review by jooseppi-luna — APPROVED** [review]
lgtm nice work!

File: samples/storage_csm_powerflex_v230.yaml [hunk 2]
```diff
@@ -36,11 +36,13 @@ spec:
     sideCars:
     # comment the following section if you don't want to run the monitoring sidecar
```
Comment (abhi16394) [inline samples/storage_csm_powerflex_v230.yaml:37]: could you remove this comment and add comment similar to storage_csm_powerflex_v240.yaml

**Review by abhi16394 — APPROVED** [review]
lgtm

**Review by jooseppi-luna — APPROVED** [review]

**Pull request merged**